Introduce $(LIBDIR) to Makefile

The variable $(LIBDIR) allows us to override where libbz2 is installed.

Index: bzip2-1.0.6/Makefile
===================================================================
--- bzip2-1.0.6.orig/Makefile	2010-09-10 15:46:02.000000000 -0700
+++ bzip2-1.0.6/Makefile	2017-11-23 12:51:53.318896836 -0800
@@ -19,6 +19,7 @@
 AR=ar
 RANLIB=ranlib
 LDFLAGS=
+LIBDIR ?= lib
 
 override CFLAGS += -Wall
 
@@ -71,7 +73,7 @@
 
 install: bzip2 bzip2recover
 	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
+	if ( test ! -d $(PREFIX)/$(LIBDIR) ) ; then mkdir -p $(PREFIX)/$(LIBDIR) ; fi
 	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
 	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
 	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
@@ -87,8 +89,8 @@
 	chmod a+r $(PREFIX)/man/man1/bzip2.1
 	cp -f bzlib.h $(PREFIX)/include
 	chmod a+r $(PREFIX)/include/bzlib.h
-	cp -f libbz2.a $(PREFIX)/lib
-	chmod a+r $(PREFIX)/lib/libbz2.a
+	cp -f libbz2.a $(PREFIX)/$(LIBDIR)
+	chmod a+r $(PREFIX)/$(LIBDIR)/libbz2.a
 	cp -f bzgrep $(PREFIX)/bin/bzgrep
 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzegrep
 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzfgrep
